File: app/src/main/java/com/example/bookapp/viewModels/ViewModelComments.java
package com.example.bookapp.viewModels;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.bookapp.AppUtilities;
import com.example.bookapp.models.Comment;
import com.example.dataLayer.dataObjectsToSerialize.SerializeComment;

import java.util.ArrayList;

public class ViewModelComments extends ViewModel {
    private CommentsRepository commentsRepository;
    private MutableLiveData<ArrayList<Comment>> currentPostComments;
    public ViewModelComments() {
        super();
        commentsRepository = CommentsRepository.getInstance(AppUtilities.getRetrofit());
    }

    public void uploadComment(@NonNull SerializeComment serializeComment) {
        commentsRepository.uploadComment(serializeComment);
    }

    public MutableLiveData<ArrayList<Comment>> getCommentsForPost(long postID) {
        currentPostComments = commentsRepository.fetchCommentsForPost(postID);

        return currentPostComments;
    }
}

File: app/src/main/java/com/example/bookapp/viewModels/ViewModelComments.kt
package com.example.bookapp.viewModels

import android.app.Application
import androidx.hilt.lifecycle.ViewModelInject
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.example.bookapp.models.Post
import com.example.dataLayer.models.PostWithComments
import com.example.dataLayer.models.serialization.SerializeComment
import com.example.dataLayer.repositories.OperationStatus
import kotlinx.coroutines.InternalCoroutinesApi
import javax.inject.Inject

@InternalCoroutinesApi
class ViewModelComments @ViewModelInject constructor(
        private val commentsRepository: CommentsRepository
) : ViewModel() {



    fun uploadComment(comment: SerializeComment): LiveData<OperationStatus> = commentsRepository.uploadComment(comment)

    fun getCommentsForPost(post: Post): LiveData<PostWithComments> = commentsRepository.getCommentsForPost(post);


}